sdk/nodejs/entrypoint/load.ts

```diff
@@ -0,0 +1,35 @@
+import { dag, ID } from "../api/client.gen.js"
+
+/**
+ * Import all given typescript files so that trigger their decorators
+ * and register their class and functions inside the Registry.
+ *
+ * @param files List of files to load.
+ */
+export async function load(files: string[]): Promise<void> {
+  await Promise.all(files.map(async (f) => await import(f)))
+}
+
+/**
+ * Load argument as Dagger type.
+ *
+ * This function remove the quote from the identifier and checks
+ * if it's a Dagger type, if it is, it loads it according to
+ * its type.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function loadArg(value: string): Promise<any> {
+  const trimmedValue = value.slice(1, value.length - 1)
+
+  const [source] = trimmedValue.split(":")
+
+  const [origin, type] = source.split(".")
+  if (origin === "core") {
+    // Workaround to call get any object that has an id
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return dag[`load${type}FromID`](trimmedValue as ID)
```

Comment:
Python does this differently. The input is only converted right before calling the function, not right after fetching the input. The important thing is to do it when you know what the type you need is. So at that time, rather than just `loadArg(val)`, you can `loadArg(val, type)` and so if for example the type is `dagger.Directory` you know to call `dag.loadDirectoryFromID` instead of reading it from the value.

## Example

First get the decoded inputs from JSON (as primitives still):

https://github.com/dagger/dagger/blob/5adee9f4b047d0f32aa4faf7839e87ad9e9570b6/sdk/python/src/dagger/mod/_module.py#L239-L250

Only later, convert values based on desired types:

https://github.com/dagger/dagger/blob/5adee9f4b047d0f32aa4faf7839e87ad9e9570b6/sdk/python/src/dagger/mod/_resolver.py#L309-L321

For core types, if you have both value and type:

https://github.com/dagger/dagger/blob/5adee9f4b047d0f32aa4faf7839e87ad9e9570b6/sdk/python/src/dagger/mod/_converter.py#L37-L47